lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -1668,87 +1628,87 @@ class SemanticsObject {
     _currentChildrenInRenderOrder = childrenInRenderOrder;
   }
 
-  /// The primary role of this node.
+  /// The role of this node.
   ///
-  /// The primary role is assigned by [updateSelf] based on the combination of
+  /// The role is assigned by [updateSelf] based on the combination of
   /// semantics flags and actions.
-  PrimaryRoleManager? primaryRole;
+  SemanticRole? semanticRole;
 
-  PrimaryRole _getPrimaryRoleIdentifier() {
+  SemanticRoleId _getSemanticRoleId() {
     // The most specific role should take precedence.
     if (isPlatformView) {
-      return PrimaryRole.platformView;
+      return SemanticRoleId.platformView;
     } else if (isHeading) {
-      return PrimaryRole.heading;
+      return SemanticRoleId.heading;
     } else if (isTextField) {
-      return PrimaryRole.textField;
+      return SemanticRoleId.textField;
     } else if (isIncrementable) {
-      return PrimaryRole.incrementable;
+      return SemanticRoleId.incrementable;
     } else if (isVisualOnly) {
-      return PrimaryRole.image;
+      return SemanticRoleId.image;
     } else if (isCheckable) {
-      return PrimaryRole.checkable;
+      return SemanticRoleId.checkable;
     } else if (isButton) {
-      return PrimaryRole.button;
+      return SemanticRoleId.button;
     } else if (isScrollContainer) {
-      return PrimaryRole.scrollable;
+      return SemanticRoleId.scrollable;
     } else if (scopesRoute) {
-      return PrimaryRole.dialog;
+      return SemanticRoleId.dialog;
     } else if (isLink) {
-      return PrimaryRole.link;
+      return SemanticRoleId.link;
     } else {
-      return PrimaryRole.generic;
+      return SemanticRoleId.generic;
     }
   }
 
-  PrimaryRoleManager _createPrimaryRole(PrimaryRole role) {
+  SemanticRole _createSemanticRole(SemanticRoleId role) {
```

Comment:
This is probably only used from within the class, but this probably should be `factory` (or at least `static`)? (Even though it needs "this" to build itself, ouch)

Reply:
Can't be static. It uses `this`. But we could revisit this. Maybe sealed classes can help clean this up.